functions/Import-DbaCsv.ps1

```diff
@@ -439,7 +439,14 @@ function Import-DbaCsv {
                 $elapsed = [System.Diagnostics.Stopwatch]::StartNew()
                 # Open Connection to SQL Server
                 try {
-                    $server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $SqlCredential -StatementTimeout 0 -MinimumVersion 9
+                    $server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $SqlCredential -Database $Database -StatementTimeout 0 -MinimumVersion 9
+
+                    # boundary case: ensure the correct db is referenced in the $server object
+                    if ($Database -ne $server.ConnectionContext.DatabaseName) {
+                        Stop-Function -Message "The SqlInstance object for $instance is currently connected to the database `"$($server.ConnectionContext.DatabaseName)`" and does not match the parameter `"-Database $Database`". If a pre-connected SqlInstance object is passed in it must be connected to the database specified by the -Database parameter."
```

Comment:
Connecting to Azure if `Connect-DbaInstance` is used to create a connection object and the database **is not provided** then the database name on that context is null. Which causes the message to not provide the current database name.

A more consistent property and one that will be populated, regardless, is `CurrentDatabase`:

```powershell
[PS 5.1] [16:39:40] [  4.0ms ][ git:pr/lancasteradam/7064 ] C:\git\dbatools
 [27] > $cn.ConnectionContext             


StatementTimeout          : 600
ConnectionType            : DatabaseEngine
BatchSeparator            : GO
TransactionDepth          : 0
SqlExecutionModes         : ExecuteSql    
FixedServerRoles          : None
UserProfile               : 
ProcessID                 : 63
TrueLogin                 : localadmin
TrueName                  : thy
InUse                     : True
LockTimeout               : -1
ServerVersion             : 12.0.2000
ProductVersion            : 12.0.2000.8
DatabaseEngineType        : SqlAzureDatabase
DatabaseEngineEdition     : SqlDatabase
HostPlatform              : Windows
ConnectionProtocol        : TcpIp
IsContainedAuthentication : False
SqlConnectionObject       : System.Data.SqlClient.SqlConnection
CurrentDatabase           : master
IsOpen                    : True
CapturedSql               : Microsoft.SqlServer.Management.Common.CapturedS
AutoDisconnectMode        : DisconnectIfPooled
IsForceDisconnected       : False
AccessToken               : 
ServerInstance            : TCP:thy.database.windows.net,1433
Login                     : localadmin
Password                  : P@ssword$01234
SecurePassword            : System.Security.SecureString
LoginSecure               : False
ConnectAsUserName         : 
ConnectAsUserPassword     : 
ConnectAsUser             : False
NonPooledConnection       : True
PooledConnectionLifetime  : 0
MinPoolSize               : 0
MaxPoolSize               : 100
ConnectTimeout            : 30
Authentication            : NotSpecified
ApplicationIntent         : 
TrustServerCertificate    : False
ConnectionString          : Data Source=TCP:thy.database.windows.net,1433;U
                            PowerShell module - dbatools.io"
SecureConnectionString    : System.Security.SecureString
NetworkProtocol           : NotSpecified
ApplicationName           : 
WorkstationId             : THY-01-0145-LT
DatabaseName              : 
PacketSize                : 8000
MultipleActiveResultSets  : False
EncryptConnection         : False
```

Reply:
```powershell
[PS 5.1] [16:41:29] [496.0ms ][ git:pr/lancasteradam/7064 ] C:\git\dbatools
 [32] > Import-DbaCsv -FullPath C:\temp\testexport.txt -SqlInstance $cn -Database ss -Table dbatoolslci_testImportprwithcnobj -Schema dbo -Truncate -AutoCreateTable -NoProgress
WARNING: [16:42:36][Import-DbaCsv] The SqlInstance object for thy.database.windows.net is currently connected to the database "" and does not match the parameter "-Database ss". If a pre-connected SqlInstance object is passed in it must 
be connected to the database specified by the -Database parameter.
```

Reply:
Thanks for catching that. I see what you mean, even the MS docs have a 'TBD' for the description of the DatabaseName property. The CurrentDatabase property on the other hand does have documentation. I'll upload that change.